lib/sugar.js

```diff
@@ -22,11 +22,14 @@ if (typeof navigator !== 'undefined') {
   var sugarClient = new SugarClient();
 
   auth.listen('change', function() {
-    Promise.all(auth.checkCurrent(), auth.checkBearerToken())
-      .then(function([ user, token ]) {
-        if (user && token) {
+    auth.checkCurrent()
+      .then(function(user) {
+        if (user) {
           sugarClient.userId = user.id;
-          sugarClient.authToken = token;
+          auth.checkBearerToken()
```

Comment:
I believe the `sugar.connect()` (below) needs to be in this promise then block to ensure the connection is after the token is setup

Reply:
this is in the wrong spot - apologies.